packages/react-app/src/contexts/BridgeContext.jsx

```diff
@@ -21,32 +21,40 @@ import {
   parseValue,
 } from 'lib/helpers';
 import { fetchTokenDetails } from 'lib/token';
+import { fetchTokenList } from 'lib/tokenList';
 import React, { useCallback, useEffect, useState } from 'react';
 
 export const BridgeContext = React.createContext({});
 
 export const BridgeProvider = ({ children }) => {
-  const { ethersProvider, account, providerChainId } = useWeb3Context();
+  const {
+    ethersProvider,
+    account,
+    providerChainId,
+    isTokenInjected,
```

Comment:
Do we need to fetch isTokenInjected from Web3Provider ? 
We could just read the token query param in BridgeProvider, what do you think ? 

Lets keep token information in BridgeContext alone.

Reply:
The main reason for this implementation is because of the uncertainty of query params' presence. Hence what Im doing is, I check first if the user has even sent a chainId at the first instance, and then process the boolean logic of valid. That eases the use case problem solving. If I hadnt used it in this way, then the app launches at first without waiting for the parsing of query params and straight away goes towards the providerChainId selection. 

But yeah, Im all ears. (Just my thoughts above)

Reply:
Let the chainId query extraction happen here itself. I'm asking you to move the token query extraction to BridgeContext

Reply:
Ahh. I see, sorry about that. Understood it now.